pywbem/cim_operations.py

```diff
@@ -954,11 +1077,81 @@ def __repr__(self):
         return "%s(url=%r, creds=%s, " \
                "default_namespace=%r, x509=%r, verify_callback=%r, " \
                "ca_certs=%r, no_verification=%r, timeout=%r, " \
-               "use_pull_operations=%r)" % \
+               "use_pull_operations=%r, stats=%r, recorder=%r)" % \
                (self.__class__.__name__, self.url, creds_repr,
                 self.default_namespace, self.x509, self.verify_callback,
                 self.ca_certs, self.no_verification, self.timeout,
-                self.use_pull_operations)
+                self.use_pull_operations, self.statistics,
+                self.operation_recorder)
+
+    def _record_wbemconnection(self, recorder):
+        """
+            Create the recorde for recording wbemconnection information
+            and forward to recorder
+        """
+        creds = '%s:%s' % (self.creds[0], '******') if self.creds else None
+        print('cimoperation creds %s' % creds)
+        recorder.reset()
+        recorder.stage_wbem_connection(
+            url=self.url, conn_id=self.conn_id,
+            creds=creds,
+            default_namespace=self.default_namespace,
+            no_verification=self.no_verification,
+            timeout=self.timeout,
+            use_pull_operations=self.use_pull_operations,
+            stats_enabled=self.stats_enabled)
+
+    def add_operation_recorder(self, operation_recorder):
+        # pylint: disable=line-too-long
+        """
+        Adds an operation recorder instance to the list of operation recorders
+        for this WBEMConnection.  This allows multiple recorders to be
+        uses simultaneously.  It ignores multiple additions of the
+        same recorder.
+
+        Parameters:
+          operation_recorder(:class:`~pywbem._recorder.TestClientRecorder or :class:`~pywbem._recorder.LogOperationRecorder)
+            Adds the instance of the defined recorder to the list of
+            active operation recorders
+
+        Exception:
+            ValueError: If the same recorder class is added multiple times.
```

Comment:
We add objects, not classes. So: "If an object of the same recorder class"

Reply:
I agree but that is sort of messy.  The phrasing does not sound good

Reply:
Referencing objects in place of instances seems to get garbled sometimes.

Mark done